app/controllers/users_controller.rb

```diff
@@ -0,0 +1,50 @@
+class UsersController < ApplicationController
+
+  get '/login' do
+    erb :"users/login"
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+
+    if @user && @user.authenticate(params[:password])
+      session[:id] = @user.id
```

Comment:
I would change your session key to be more explicit, as you are storing a user id it would make more sense for this to be `session[:user_id]`